apps/v1/register.go

```diff
@@ -1,50 +1,37 @@
 package v1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
 const (
-	LegacyGroupName = ""
-	GroupName       = "apps.openshift.io"
+	Group = "apps.openshift.io"
 )
 
 var (
-	SchemeGroupVersion       = schema.GroupVersion{Group: GroupName, Version: "v1"}
-	LegacySchemeGroupVersion = schema.GroupVersion{Group: LegacyGroupName, Version: "v1"}
-
-	LegacySchemeBuilder    = runtime.NewSchemeBuilder(addLegacyKnownTypes)
-	AddToSchemeInCoreGroup = LegacySchemeBuilder.AddToScheme
-
-	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
-	AddToScheme   = SchemeBuilder.AddToScheme
+	// generated code relies on this name
+	SchemeGroupVersion = Version
+	Version            = schema.GroupVersion{Group: Group, Version: "v1"}
```

Comment:
when referenced elsewhere, it's unclear whether this is a GroupVersion or a Version

Reply:
> when referenced elsewhere, it's unclear whether this is a GroupVersion or a Version

Should we encourage the logical separate of those things?

Reply:
The answer to "what version is apps/v1" could reasonably be either "v1" or {apps, v1}, depending on what you meant. I think naming it GroupVersion is clearer… is there a downside?

Reply:
> The answer to "what version is apps/v1" could reasonably be either "v1" or {apps, v1}, depending on what you meant. I think naming it GroupVersion is clearer… is there a downside?

You eventually end up with Germans (and others) complaining about name length.

@sttts preference one way or the other?

Reply:
GroupVersion

Reply:
> The answer to "what version is apps/v1" could reasonably be either "v1" or {apps, v1}, depending on what you meant. I think naming it GroupVersion is clearer… is there a downside?

Also, we don't generally name our variables after types and instead name them after meaning.  This means a version and for the vast majority of developers a non-group qualified version isn't a valuable thing.

Reply:
How does `Version.Version` look like? Odd?

Reply:
🍿

Reply:
> You eventually end up with Germans (and others) complaining about name length.

Since others were brought to game I chime in and will vote for `GroupVersion` as well. I prefer verbosiness over shortness.